robottelo/ui/hosts.py

```diff
@@ -8,138 +8,193 @@
-            self.wait_until_element(tab_locators["host.tab_os"]).click()
-            type_arch = self.find_element(locators["host.arch"])
-            Select(type_arch).select_by_visible_text(arch)
-            self.wait_for_ajax()
-            type_os = self.find_element(locators["host.os"])
-            Select(type_os).select_by_visible_text(os)
-            self.wait_for_ajax()
-            type_media = self.find_element(locators["host.media"])
-            Select(type_media).select_by_visible_text(media)
-            self.wait_for_ajax()
-            type_ptable = self.find_element(locators["host.ptable"])
-            Select(type_ptable).select_by_visible_text(ptable)
-            self.wait_for_ajax()
-            if custom_ptable:
-                custom = self.find_element(locators["host.custom_ptables"])
-                custom.send_keys(custom_ptable)
-            password = self.find_element(locators["host.root_pass"])
-            password.send_keys(root_pwd)
-            self.find_element(locators["host.provision_template"]).click()
+            Select(
+                self.wait_until_element(locators['host.os'])
+            ).select_by_visible_text(os)
             self.wait_for_ajax()
+        if media:
+            Select(
+                self.wait_until_element(locators['host.media'])
+            ).select_by_visible_text(media)
+        if ptable:
+            Select(
+                self.wait_until_element(locators['host.ptable'])
+            ).select_by_visible_text(ptable)
+        if custom_ptable:
+            self.wait_until_element(
+                locators['host.custom_ptables']).send_keys(custom_ptable)
+        if root_pwd:
+            self.wait_until_element(
+                locators['host.root_pass']).send_keys(root_pwd)
 
-    def create(self, name, domain, subnet, org, host_group=None, resource=None,
-               env=None, ip_addr=None, mac=None, os=None, arch=None,
-               media=None, ptable=None, custom_ptable=None, root_pwd=None,
-               cpus="1", memory="768 MB"):
+    def create(self, arch=None, cpus='1', cv=None, custom_ptable=None,
+               domain=None, env=None, host_group=None, ip_addr=None,
+               lifecycle_env=None, loc=None, mac=None, media=None,
+               memory='768 MB', name=None, org=None, os=None, ptable=None,
+               puppet_ca=None, puppet_master=None, resource=None,
+               root_pwd=None, subnet=None):
         """Creates a host."""
-        self.wait_until_element(locators["host.new"]).click()
-        if self.wait_until_element(locators["host.name"]):
-            self.find_element(locators["host.name"]).send_keys(name)
+        if (
+                not arch or
+                not domain or
+                not env or
+                not mac or
+                not media or
+                not name or
+                not os or
+                not ptable or
+                not root_pwd):
+            raise UIError(
```

Comment:
The same time that doing a validation is good, it may not allow you to do a negative test, for example ensuring that the UI raises an error if no root password is provided. With that said, I think is better to not do the validation as it will allow create a more complete set of tests.